tests/llm/test_llm_functions.py

```diff
@@ -93,3 +99,139 @@ def test_assistant_functions():
+
+    # different base ids, longest element
+    features = ["id2", "id2;id3"]
+    get_annotation_from_uniprot_by_feature_list(features)
+    mock_uniprot_annotation.assert_called_with("id2;id3")
+
+    features = []
+    with pytest.raises(ValueError, match="No features provided"):
+        get_annotation_from_uniprot_by_feature_list(features)
+
+
+@patch("alphastats.llm.llm_functions.format_uniprot_annotation")
+@patch("alphastats.llm.llm_functions.get_annotation_from_uniprot_by_feature_list")
+@patch("alphastats.llm.llm_functions.get_annotation_from_store_by_feature_list")
+@patch("streamlit.session_state", new_callable=dict)
+def test_get_uniprot_info_for_llm_input(
+    mock_session_state,
+    mock_get_annotation_from_store,
+    mock_get_annotation_from_uniprot,
+    mock_format_uniprot_annotation,
+):
+    """Test that the function retrieves the correct UniProt information for the LLM input."""
+
+    class DUMMY_DATASET:
+        _feature_to_repr_map = {
+            "id1;id4": "gene2;gene4",
+            "id2": "gene2",
+            "id3": "gene1",
+            "id5;id1": "ids:id5",
+        }
+        _gene_to_features_map = {
+            "gene1": ["id3"],
+            "gene2": ["id1;id4", "id2"],
+            "gene4": ["id1;id4"],
+        }
+        _protein_to_features_map = {
+            "id1": ["id1;id4", "id5;id1"],
+            "id2": ["id2"],
+            "id3": ["id3"],
+            "id4": ["id1;id4"],
+            "id5": ["id5;id1"],
+        }
+
+    mock_session_state[StateKeys.DATASET] = DUMMY_DATASET()
+    mock_session_state[StateKeys.ANNOTATION_STORE] = {}
+    mock_session_state[StateKeys.SELECTED_UNIPROT_FIELDS] = []
+
+    # repr
+    llm_input = "gene2;gene4"
+    mock_get_annotation_from_store.return_value = None
+    mock_get_annotation_from_uniprot.return_value = {"annotation": "annotation1"}
+    mock_format_uniprot_annotation.return_value = ""
+    result = get_uniprot_info_for_llm_input(llm_input)
+    mock_get_annotation_from_store.assert_called_with(["id1;id4"], {})
+    mock_get_annotation_from_uniprot.assert_called_with(["id1;id4"])
+    assert result == "gene2;gene4: "
+
+    mock_get_annotation_from_store.return_value = {"annotation": "annotation1"}
+    # repr that is also gene
+    llm_input = "gene2"
+    get_uniprot_info_for_llm_input(llm_input)
+    mock_get_annotation_from_store.assert_called_with(["id2"], {})
+    mock_get_annotation_from_uniprot.assert_called_once()
+
+    # feature id
+    llm_input = "id1;id4"
+    get_uniprot_info_for_llm_input(llm_input)
+    mock_get_annotation_from_store.assert_called_with(["id1;id4"], {})
+
+    # gene
+    llm_input = "gene4"
+    get_uniprot_info_for_llm_input(llm_input)
+    mock_get_annotation_from_store.assert_called_with(["id1;id4"], {})
+
+    # protein
+    llm_input = "id1"
+    get_uniprot_info_for_llm_input(llm_input)
+    mock_get_annotation_from_store.assert_called_with(["id1;id4", "id5;id1"], {})
+
+    # not valid
+    llm_input = "id6"
```

Comment:
this could be split into ~6 different tests (using `DUMMY_DATASET()` as a fixture)